src/adhocracy_core/adhocracy_core/schema/__init__.py

```diff
@@ -474,8 +475,7 @@ def _deserialize_location_or_url(self, node, value):
             application_url_len = len(request.application_url)
             if application_url_len > len(str(value)):
                 raise KeyError
-            # Fixme: This does not work with :term:`virtual hosting`
-            path = value[application_url_len:]
+            path = urllib.parse.urlparse(value).path
             return find_resource(request.root, path)
```

Comment:
The Fixme is not solved,  virtual hosting is still broken: http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/vhosting.html